twa

```diff
@@ -118,31 +122,50 @@ function stage_0_has_https {
   # First, just probe the domain to see if it listens on 443.
   if ! nc -z -w2 "${domain}" 443 2>/dev/null; then
     FATAL "expected port 443 to be open, but it isn't"
-    no_https=1 ; return
+    no_https=1
   fi
 
-  # The rest below would be cool, but is probably unnecessary since `curl`
-  # won't connect to an insecure HTTPS server by default.
-  # # Then, use `openssl` to retrieve the certificate and test its validity.
-  # cert=$(openssl s_client -connect "${domain}:443" < /dev/null)
+  fetch_secure -o /dev/null "https://${domain}" 2>/dev/null
```

Comment:
The tests here and below are nice from a reporting perspective, but I originally left them out because `curl` itself should be able to do all of this verification internally (checking cert trust, whether expired, etc) -- we just need to find a way to expose it cleanly.

Here's the high-level picture I had in mind for stage 0:

1. (Already done). Test to see if the server is even listening on `443`. If it isn't, fail with `FATAL` immediately.
2. If the server *is* listening on `443`, hit it with a plain old (secure by default) `curl` call. If that `curl` fails for any reason, fail with `FATAL` immediately.
3. If the `curl` call succeeds, then do some certificate probing (whether it's a high-quality CA, whether the server accepts a reasonable cipher suite, etc).

Some gotchas: even with an explicit `https://` url, `curl` will follow an insecure redirect with `-L` (e.g., `https://example.com` -> `http://example.com`). This should *not* be recorded as a success, even though an HTTPS connection *was* successfully established. According to https://github.com/curl/curl/issues/226#issuecomment-273890336, the right way to prevent this is with `--proto-redir =https`